build.sbt

```diff
@@ -462,6 +476,210 @@ lazy val spark = (project in file("spark"))
+      "org.apache.spark" %% "spark-catalyst" % sparkVersion.value % "provided",
+
+      // Test dependencies
+      "org.junit.jupiter" % "junit-jupiter-api" % "5.8.2" % "test",
+      "org.junit.jupiter" % "junit-jupiter-engine" % "5.8.2" % "test",
+      "org.junit.jupiter" % "junit-jupiter-params" % "5.8.2" % "test",
+      "net.aichler" % "jupiter-interface" % "0.11.1" % "test",
+      // Spark test classes for Scala/Java test utilities
+      "org.apache.spark" %% "spark-catalyst" % sparkVersion.value % "test" classifier "tests",
+      "org.apache.spark" %% "spark-core" % sparkVersion.value % "test" classifier "tests",
+      "org.apache.spark" %% "spark-sql" % sparkVersion.value % "test" classifier "tests",
+      // ScalaTest for test utilities (needed by Spark test classes)
+      "org.scalatest" %% "scalatest" % scalaTestVersion % "test"
+    ),
+    Test / testOptions += Tests.Argument(TestFrameworks.JUnit, "-v", "-a")
+  )
+
+
+// ============================================================
+// Module 4: delta-spark (final published module - combined v1+v2)
+// ============================================================
+lazy val spark = (project in file("spark-combined"))
+  .dependsOn(sparkV1)
+  .dependsOn(sparkV2)
+  .dependsOn(storage)
+  .disablePlugins(JavaFormatterPlugin, ScalafmtPlugin)
+  .settings (
+    name := "delta-spark",
+    commonSettings,
+    scalaStyleSettings,
+    sparkMimaSettings,
+    releaseSettings, // Published to Maven as delta-spark.jar
+    
+    // Set Test baseDirectory before crossSparkSettings() so it uses the correct directory
+    Test / baseDirectory := (sparkV1 / baseDirectory).value,
+    
+    // Test sources from spark/ directory (sparkV1's directory)
+    // MUST be set BEFORE crossSparkSettings() to avoid overwriting version-specific directories
+    Test / unmanagedSourceDirectories := {
+      val sparkDir = (sparkV1 / baseDirectory).value
+      Seq(
+        sparkDir / "src" / "test" / "scala",
+        sparkDir / "src" / "test" / "java"
+      )
+    },
+    Test / unmanagedResourceDirectories := Seq(
+      (sparkV1 / baseDirectory).value / "src" / "test" / "resources"
+    ),
+    
+    crossSparkSettings(),
+    
+    // MiMa should use the generated JAR (not classDirectory) because we merge classes at package time
+    mimaCurrentClassfiles := (Compile / packageBin).value,
+    
+    // Export as JAR to dependent projects (e.g., connectServer, connectClient).
+    // This prevents classpath conflicts from internal module 'classes' directories.
+    exportJars := true,
+
+    // Internal module artifact names to exclude from published POM
+    internalModuleNames := Set("delta-spark-v1", "delta-spark-v1-shaded", "delta-spark-v2"),
+
+    // Merge classes from internal modules (v1, v2, storage) into final JAR
+    // kernel modules are kept as separate JARs and listed as dependencies in POM
+    Compile / packageBin / mappings ++= {
```

Comment:
Can you test this in another PR? i.e. checkout -b a new branch, and then chagne some other build.sbt code such that this check DETECTS it and fails? and push that PR? and we can link to it in this PR description? and then we have positive evidence that this detection works?